screenpy/protocols.py

```
@@ -21,6 +21,8 @@
 class Answerable(Protocol):
     """Questions are Answerable"""
 
+    caught_exception: Any
```

Should we add a comment here to explain that the `caught_exception` is used for logging, if the `Answerable` handles exceptions?

---

I had another thought—this should probably be typed `Exception` instead of `Any`.

---

I thought about typing an exception. Technically it only really needs to be an object that supports `__str__`.  Under typical conditions I would fully expect it to be an `Exception`.  But I've been a little gunshy with restricting types via typehints lately.  I keep running into situations where I need to adjust the hint for a slightly wider type.  

My fear is some question will want to store an error string (or some other type of message object) without having caught an exception.  Am I overthinking it?

---

I think with a name like `caught_exception`, it should probably only ever be an `Exception`. I was going to say that we're relying on it to be an exception, but you're right, we're only appending it to the logged message right now. 

Maybe we'd want to re-throw the exception later for some other interaction, though, so i'd really like to expect this to be an Exception. If someone needs it to be something else for their own custom Action or something else, they can add a different attribute. :P

Hey by the way, does adding this to the `Protocol` make our existing Questions incompatible if they don't have a `caught_exception` defined?

---

I can add a comment here. I felt I would have to start explaining how hamcrest utilizes the reason argument in `assert_that` for things to make sense.

---

> does adding this to the Protocol make our existing Questions incompatible
**Good catch.**  mypy balks when a question doesn't set `caught_exception` to something.  

a) I will add unit tests to verify objects conform to their protocols.  
b) the commit hooks didn't catch it because we don't require mypy to pass on the unit tests themselves. *derp
c) I'm not sure what we want to do about this.  I need to ponder for a bit.  (might see if your around for a brainstorming session to discuss)

---

See Latest commits.